fog/test-client/src/config.rs

```diff
@@ -1,19 +1,33 @@
 // Copyright (c) 2018-2021 The MobileCoin Foundation
 
+//! Configuration parameters for the test client binary
+
 use mc_common::logger::{log, Logger};
-use mc_util_uri::ConsensusClientUri;
+use mc_fog_uri::{FogLedgerUri, FogViewUri};
+use mc_util_uri::{AdminUri, ConsensusClientUri};
 
 use mc_fog_sample_paykit::AccountKey;
 
+use serde::Serialize;
 use std::{path::PathBuf, str::FromStr, time::Duration};
 use structopt::StructOpt;
 
-pub const TEST_FOG_AUTHORITY_FINGERPRINT: [u8; 4] = [9, 9, 9, 9];
-pub const TEST_FOG_REPORT_ID: &str = "";
-
-#[derive(Debug, StructOpt)]
+/// StructOpt for test-client binary
+#[derive(Debug, StructOpt, Serialize, Clone)]
 #[structopt(name = "test-client", about = "Test client for Fog infrastructure.")]
-pub struct Config {
+pub struct TestClientConfig {
+    /// A URI to host the prometheus data at.
+    ///
+    /// If provided, then we continuously send test transfers and publish
+    /// metrics, via the grpc admin API endpoint hosted at this URI.
+    ///
+    /// The frequency of transactions can be configured with "transfer_period".
+    ///
+    /// When running continuously, num_transactions is ignored, and we do not
+    /// fail fast when deadlines are exceeded.
+    #[structopt(long)]
+    pub admin_listen_uri: Option<AdminUri>,
```

Comment:
I think piggybacking the continuous mode on this flag is a bit hacky. I would prefer a `--continuous` flag that controls this, and the admin interface becomes optional.